src/machines/confirmation-dialog-machine.ts

```diff
@@ -0,0 +1,182 @@
+// Source: https://xstate-catalogue.com/machines/confirmation-dialog
+import {assign, createMachine} from 'xstate'
+import {Dispatch, SetStateAction} from 'react'
+import {getAuthorizationHeader} from 'utils/auth'
+import axios from 'axios'
+import toast from 'react-hot-toast'
+
+type Member = {
+  id: number
+  name: string
+  email: string
+}
+
+export interface ConfirmationDialogMachineContext {
+  action?: () => Promise<void>
+  errorMessage?: string
+  accountId: number | undefined
+  memberToRemove: Member | undefined
+  setMembers: Dispatch<SetStateAction<any[]>>
+}
+
+export type ConfirmationDialogMachineEvent =
+  | {
+      type: 'OPEN_DIALOG'
+      action: () => Promise<void>
+      payload: {member: Member}
+    }
+  | {
+      type: 'CONFIRM'
+    }
+  | {
+      type: 'CANCEL'
+    }
+
+const removeTeamMember = (
+  accountId: number | undefined,
+  userId: number | undefined,
+) => {
+  if (accountId && userId) {
+    const removeTeamMemberUrl = `${process.env.NEXT_PUBLIC_AUTH_DOMAIN}/api/v1/accounts/${accountId}/team_members/${userId}`
+
+    return axios.delete(removeTeamMemberUrl, {
+      headers: {...getAuthorizationHeader()},
+    })
+  }
+}
+
+const confirmationDialogMachine = createMachine<
+  ConfirmationDialogMachineContext,
+  ConfirmationDialogMachineEvent
+>(
+  {
+    id: 'confirmationDialog',
+    initial: 'closed',
+    context: {
+      accountId: undefined,
+      memberToRemove: undefined,
+      setMembers: () => {},
+    },
+    states: {
+      closed: {
+        id: 'closed',
```

Comment:
is there a reason you us the state id here? since the name of the state is `closed` couldnt we omit the id and just use `closed` instead of `#closed` in the actions?

Reply:
This is another case of, "this is what the template gave me." It seems likely that I could make the adjustment you are suggesting.

<img width="617" alt="CleanShot 2021-04-27 at 15 39 28@2x" src="https://user-images.githubusercontent.com/694063/116309850-e418be00-a76e-11eb-9656-1cdfb9c88bb8.png">

Reply:
Interesting... so when I remove the `id: 'closed'` and reference that state directly, I get the following error when trying to transition out of the `open.idle` state with `CANCEL`. So perhaps when transitioning between levels of nesting the `id` is needed? 🤔 

<img width="990" alt="CleanShot 2021-04-27 at 17 36 55@2x" src="https://user-images.githubusercontent.com/694063/116321234-51344f80-a77f-11eb-930c-a9cd3ddc5cb3.png">

Reply:
That's correct - the id is necessary to route you from the nested state to a sibling of its parent